src/__tests__/LoadingOverlay.test.tsx

```
@@ -0,0 +1,342 @@
+        </div>
+      );
+      
+      expect(screen.getByTestId('parent')).toBeInTheDocument();
+      expect(screen.getByTestId('sibling')).toBeInTheDocument();
+      expect(screen.getByRole('presentation')).toBeInTheDocument();
+    });
+
+    it('should overlay content correctly when loading', () => {
+      render(
+        <div>
+          <button>Click me</button>
```

Biome (1.9.4)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)